src/dkg.ts

```diff
@@ -122,15 +122,25 @@ export class DkgClient {
     );
 
     if (waitUntilEnd) {
-      const isSuccessful = await DkgClient.waitUntilRitualEnd(
-        web3Provider,
-        ritualId
-      );
-      if (!isSuccessful) {
+      const timeout = await DkgCoordinatorAgent.getTimeout(web3Provider);
```

Comment:
Haven't thought this through fully just yet, but you may be able to leverage the Coordinator contract to do the bulk of the timeout work for you. See https://github.com/nucypher/nucypher-contracts/blob/main/contracts/contracts/coordination/Coordinator.sol#L75.

The ritual state will return TIMEOUT if the Ritual has not completed within the timeout window. Of course, you don't want to loop hitting the Coordinator contract, so only hitting it periodically could be simpler.

Basically the end case for waiting is you receive `EndRitual` or the returned state from the Coordinator contract is INVALID / TIMEOUT / FINALIZED.